src/devices/Vl53L0X/Vl53L0X.cs

```diff
@@ -0,0 +1,1163 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// This code has been ported from the Dexter Industries Python code
+// https://github.com/DexterInd/DI_Sensors/blob/master/Python/di_sensors/VL53L0X.py
+// It is based as well on the offical ST Microelectronics API in C
+// https://www.st.com/content/st_com/en/products/embedded-software/proximity-sensors-software/stsw-img005.html
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+
+namespace Iot.Device.Vl53L0X
+{
+    public class Vl53L0X : IDisposable
+    {
+        /// <summary>
+        /// The default I2C Address
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x29;
+        // Default address can be found in documentation
+        // page 18 with value 0x52 >> 1 = 0x29
+        private readonly I2cDevice _i2cDevice;
+        private readonly bool _autoDisposable;
+        private byte _stopData;
+        private readonly int _operationTimeout;
+        private UInt32 _measurementTimingBudgetMicrosecond;
+        private bool _continuousInitialized = false;
+        private bool _highResolution;
+        private Precision _precision;
+
+        /// <summary>
+        /// Get the sensor information including internal signal and distance offsets
+        /// </summary>
+        public Information Information { get; internal set; }
+
+        /// <summary>
+        /// Used to find a clean measurement when reading in single shot 
+        /// </summary>
+        public int MaxTryReadSingle { get; set; }
+
+        /// <summary>
+        /// Set/Get high resolution measurement
+        /// </summary>
+        public bool HighResolution
+        {
+            get
+            {
+                return _highResolution;
```

Comment:
does it make sense to not have any cache and read this value from register?

Reply:
Yes, in this situation. At initialization the register is changed and the internal value is set. Later, when you change again the resolution, same thing, the internal value is updated. Looks like there is no other way to change the resolution than writting them. 
And as the documentation is not perfect, the reference I found is writting it and not reading the register.
In all the cases, the main code will be the master on this sensor and it won't be changed by anything else. 
With all that, I think it's the best way to handle this.